config/config.go

```diff
@@ -277,7 +278,7 @@ const (
 	redisPasswordEnv = "SWARM_REDIS_PASSWORD"
 )
 
-func NewConfig() *Config {
+func NewConfig(progname string) *Config {
```

Comment:
This would be an API change.
Lets use empty name and then call https://pkg.go.dev/flag#FlagSet.Init later in Parse (before `c.Flags.Parse(args)`)